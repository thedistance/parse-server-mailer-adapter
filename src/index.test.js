/*
 * @Author: Ben Briggs <benbriggs>
 * @Date:   2018-02-26T09:46:37+00:00
 * @Email:  ben.briggs@thedistance.co.uk
 * @Last modified by:   benbriggs
 * @Last modified time: 2018-02-27T09:23:15+00:00
 * @Copyright: The Distance
 */

import adapter from '.';

const createParseObject = props => ({
  get(key) {
    return props[key];
  },
});

const getMessageParameters = ({ message }) =>
  Promise.resolve(message)
    .then(JSON.parse)
    .then(result => {
      const { from, to, subject, text, html, messageId } = result;
      return { from, to, subject, text, html, messageId };
    });

test('should send email', () => {
  const { sendMail } = adapter({
    from: 'mailserver@example.com',
    transport: { jsonTransport: true },
  });

  return sendMail({
    to: 'client@example.com',
    subject: 'Test message',
    text: 'Hi there',
  })
    .then(getMessageParameters)
    .then(({ from, to, subject, text, messageId }) => {
      expect(from).toMatchSnapshot();
      expect(to).toMatchSnapshot();
      expect(subject).toMatchSnapshot();
      expect(text).toMatchSnapshot();
      expect(messageId).toEqual(expect.any(String));
    });
});

test('should send email with names', () => {
  const { sendMail } = adapter({
    from: 'Test App <mailserver@example.com>',
    transport: { jsonTransport: true },
  });

  return sendMail({
    to: 'Client <client@example.com>',
    subject: 'Test message',
    text: 'Hi there',
  })
    .then(getMessageParameters)
    .then(({ from, to }) => {
      expect(from).toMatchSnapshot();
      expect(to).toMatchSnapshot();
    });
})

test('should handle multiple email addresses', () => {
  const { sendMail } = adapter({
    from: 'mailserver@example.com',
    transport: { jsonTransport: true },
  });

  return sendMail({
    to: ['client@example.com', 'client2@example.com'],
    subject: 'Test message',
    text: 'Hi there',
  })
    .then(getMessageParameters)
    .then(({ from, to, subject, text, messageId }) => {
      expect(from).toMatchSnapshot();
      expect(to).toMatchSnapshot();
      expect(subject).toMatchSnapshot();
      expect(text).toMatchSnapshot();
      expect(messageId).toEqual(expect.any(String));
    });
});

test('should send password reset email', () => {
  const { sendPasswordResetEmail } = adapter({
    from: 'mailserver@example.com',
    transport: { jsonTransport: true },
  });

  return sendPasswordResetEmail({
    appName: 'TestApp',
    user: createParseObject({
      email: 'client@example.com',
    }),
    link: 'https://example.com/reset-password',
  })
    .then(getMessageParameters)
    .then(({ from, to, subject, text, html, messageId }) => {
      expect(from).toMatchSnapshot();
      expect(to).toMatchSnapshot();
      expect(subject).toMatchSnapshot();
      expect(text).toMatchSnapshot();
      expect(html).toMatchSnapshot();
      expect(messageId).toEqual(expect.any(String));
    });
});

test('should send verification email', () => {
  const { sendVerificationEmail } = adapter({
    from: 'mailserver@example.com',
    transport: { jsonTransport: true },
  });

  return sendVerificationEmail({
    appName: 'TestApp',
    user: createParseObject({
      username: 'client@example.com',
    }),
    link: 'https://example.com/confirm-account',
  })
    .then(getMessageParameters)
    .then(({ from, to, subject, text, html, messageId }) => {
      expect(from).toMatchSnapshot();
      expect(to).toMatchSnapshot();
      expect(subject).toMatchSnapshot();
      expect(text).toMatchSnapshot();
      expect(html).toMatchSnapshot();
      expect(messageId).toEqual(expect.any(String));
    });
});
