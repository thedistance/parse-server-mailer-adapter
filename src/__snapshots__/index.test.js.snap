// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle multiple email addresses 1`] = `
Object {
  "address": "mailserver@example.com",
  "name": "",
}
`;

exports[`should handle multiple email addresses 2`] = `
Array [
  Object {
    "address": "client@example.com",
    "name": "",
  },
  Object {
    "address": "client2@example.com",
    "name": "",
  },
]
`;

exports[`should handle multiple email addresses 3`] = `"Test message"`;

exports[`should handle multiple email addresses 4`] = `"Hi there"`;

exports[`should send email 1`] = `
Object {
  "address": "mailserver@example.com",
  "name": "",
}
`;

exports[`should send email 2`] = `
Array [
  Object {
    "address": "client@example.com",
    "name": "",
  },
]
`;

exports[`should send email 3`] = `"Test message"`;

exports[`should send email 4`] = `"Hi there"`;

exports[`should send email with names 1`] = `
Object {
  "address": "mailserver@example.com",
  "name": "Test App",
}
`;

exports[`should send email with names 2`] = `
Array [
  Object {
    "address": "client@example.com",
    "name": "Client",
  },
]
`;

exports[`should send password reset email 1`] = `
Object {
  "address": "mailserver@example.com",
  "name": "",
}
`;

exports[`should send password reset email 2`] = `
Array [
  Object {
    "address": "client@example.com",
    "name": "",
  },
]
`;

exports[`should send password reset email 3`] = `"Password reset request for TestApp"`;

exports[`should send password reset email 4`] = `
"Hello,

You've requested a password reset for TestApp.

To reset your password, please click the link below.

https://example.com/reset-password [https://example.com/reset-password]

If you didn't request this email you may safely ignore it.

Thanks,

The TestApp Team"
`;

exports[`should send password reset email 5`] = `
"
<p>Hello,</p>
<p>You've requested a password reset for TestApp.</p>
<p>To reset your password, please click the link below.</p>
<p><a href=\\"https://example.com/reset-password\\">https://example.com/reset-password</a></p>
<p>If you didn't request this email you may safely ignore it.</p>
<p>Thanks,</p>
<p>The TestApp Team</p>"
`;

exports[`should send verification email 1`] = `
Object {
  "address": "mailserver@example.com",
  "name": "",
}
`;

exports[`should send verification email 2`] = `
Array [
  Object {
    "address": "client@example.com",
    "name": "",
  },
]
`;

exports[`should send verification email 3`] = `"Please confirm your email address for TestApp"`;

exports[`should send verification email 4`] = `
"Hello,

You've received this message because your email address has been registered with
TestApp. Please confirm your email by clicking the link below.

https://example.com/confirm-account [https://example.com/confirm-account]

Thanks,

The TestApp Team"
`;

exports[`should send verification email 5`] = `
"
<p>Hello,</p>
<p>
  You've received this message because your email address has been registered
  with TestApp. Please confirm your email by clicking the link below.
</p>
<p><a href=\\"https://example.com/confirm-account\\">https://example.com/confirm-account</a></p>
<p>Thanks,</p>
<p>The TestApp Team</p>"
`;
